ding/reward_model/curiosity_reward_model.py

```diff
@@ -0,0 +1,201 @@
+            - pred_next_state_feature (:obj:`torch.Tensor`):
+                Run with the encoder and residual network. Return the predicted next_state's embedded feature.
+            - pred_action_logit (:obj:`Dict`):
+                Run with the encoder. Return the predicted action logit.
+        """
+        state = torch.FloatTensor(state)
+        next_state = torch.FloatTensor(next_state)
+        action_long = torch.LongTensor(action_long)
+        action = one_hot(action_long, num=self.action_shape).squeeze(1)
+        encode_state = self.feature(state)
+        encode_next_state = self.feature(next_state)
+        # get pred action logit
+        pred_action_logit = torch.cat((encode_state, encode_next_state), 1)
+        pred_action_logit = self.inverse_net(pred_action_logit)
+        # ---------------------
+
+        # get pred next state
+        pred_next_state_feature_orig = torch.cat((encode_state, action), 1)
+        pred_next_state_feature_orig = self.forward_net_1(pred_next_state_feature_orig)
+
+        # residual
+        for i in range(4):
+            pred_next_state_feature = self.residual[i * 2](torch.cat((pred_next_state_feature_orig, action), 1))
+            pred_next_state_feature_orig = self.residual[i * 2 + 1](
+                torch.cat((pred_next_state_feature, action), 1)
+            ) + pred_next_state_feature_orig
+        pred_next_state_feature = self.forward_net_2(torch.cat((pred_next_state_feature_orig, action), 1))
+        real_next_state_feature = encode_next_state
+        return real_next_state_feature, pred_next_state_feature, pred_action_logit
+
+
+@REWARD_MODEL_REGISTRY.register('curiosity')
+class ICMRewardModel(BaseRewardModel):
+    config = dict(
+        type='curiosity',
+        intrinsic_reward_type='add',
+        learning_rate=1e-3,
+        # obs_shape=6,
+        action_shape=7,
+        batch_size=64,
+        hidden_size_list=[64, 64, 128],
+        update_per_collect=100,
+        reverse_scale=1,
+    )
+
+    def __init__(self, config: EasyDict, device: str, tb_logger: 'SummaryWriter') -> None:  # noqa
+        super(ICMRewardModel, self).__init__()
+        self.cfg = config
+        assert device == "cpu" or device.startswith("cuda")
+        self.device = device
+        self.tb_logger = tb_logger
+        self.reward_model = ICMNetwork(config.obs_shape, config.hidden_size_list, config.action_shape)
+        self.reward_model.to(self.device)
+        self.intrinsic_reward_type = config.intrinsic_reward_type
+        assert self.intrinsic_reward_type in ['add', 'new', 'assign']
+        self.train_data = []
+        self.train_states = []
+        self.train_next_states = []
+        self.train_actions = []
+        self.opt = optim.Adam(self.reward_model.parameters(), config.learning_rate)
+        self.ce = nn.CrossEntropyLoss(reduction="mean")
+        self.forward_mse = nn.MSELoss(reduction='none')
```

Comment:
why use `reduction='none'`, `reduction='mean'` is enough here, you don't need to do mean operation in the following related codes